packages/oui-datagrid/src/datagrid.controller.js

```diff
@@ -0,0 +1,206 @@
+                } else {
+                    this.scrollBegin = true;
+                }
+
+                if (panel.scrollLeft <= stickyBorderWidth) {
+                    this.scrollEnd = false;
+                } else {
+                    this.scrollEnd = true;
+                }
+            });
+        };
+    }
+
+    $onInit () {
+        this.hasActionMenu = false;
+        this.scrollBegin = false;
+        this.scrollEnd = false;
+        this.firstLoading = true;
+        this.pageSize = parseInt(this.pageSize, 10) || this.config.pageSize;
+    }
+
+    $postLink () {
+        this.$compile(template)(this.$scope, (clone) => {
+            this.$element.append(clone);
+        });
+
+        const originalContent = angular.element(this.$element.data("originalContent"));
+        const columnElements = DatagridController.filterElements(originalContent, "oui-column");
+        const actionColumnElements = DatagridController.filterElements(originalContent, "oui-action-menu");
+
+        const builtColumns = this.ouiDatagridColumnBuilder.build(columnElements, this.$scope);
+
+        if (actionColumnElements.length) {
+            builtColumns.columns.push(this.ouiDatagridColumnBuilder.buildActionColumn(actionColumnElements[0]));
+            this.hasActionMenu = true;
+        }
+
+        this.columns = builtColumns.columns;
+
+        if (this.rowsLoader) {
+            this.paging = this.ouiDatagridPaging.createRemote(this.columns, builtColumns.currentSorting, this.pageSize, this.rowLoader, this.rowsLoader);
+            this.refreshData(() => this.paging.setOffset(1));
+        } else {
+            this.paging = this.ouiDatagridPaging.createLocal(this.columns, builtColumns.currentSorting, this.pageSize, this.rowLoader, this.rows);
+
+            if (this.rows) {
+                this.refreshData(() => this.paging.setRows(this.rows));
+            }
+        }
+
+        // Manage responsiveness
+        this.scrollablePanel = this.$element[0].querySelector(".oui-datagrid-responsive-container__overflow-container");
+        if (this.scrollablePanel) {
+            angular.element(this.$window).on("resize", this.checkScroll);
+            angular.element(this.scrollablePanel).on("scroll", this.checkScroll);
+        }
+    }
+
+    $doCheck () {
+        if (!angular.equals(this.previousRows, this.rows)) {
+            this.previousRows = angular.copy(this.rows);
+
+            if (this.rows && this.paging) {
+                this.refreshData(() => this.paging.setRows(this.rows));
+            }
+        }
+    }
+
+    $onDestroy () {
+        angular.element(this.$window).off("resize", this.checkScroll);
+        angular.element(this.scrollablePanel).off("scroll");
+    }
+
+    getParentScope () {
+        return this.$scope.$parent;
+    }
+
+    hasProperty (obj, prop) { // eslint-disable-line
+        if (!obj) {
+            return false;
+        }
+
+        return hasProperty(obj, prop);
+    }
+
+    // TODO
```

Comment:
We can remove todo and implement something when required

Reply:
Done